### ISSUE TITLE
Stream got blocked, or terminated.

### ISSUE THREAD

**Issue by johnadvincula-ada** [hollodotme/fast-cgi-client#68]
Hi, we have a application deployed in AWS lamda using Laravel vapor.
There are times where getting an error " Uncaught hollodotme\FastCGI\Exceptions\ReadFailedException: Stream got blocked, or terminated. in /var/task/vendor/hollodotme/fast-cgi-client/src/Sockets/Socket.php:564"

May we know how to deal with that error and possible cause?

Please see the logs.txt as reference

Thanks a lot
[logs.txt](https://github.com/hollodotme/fast-cgi-client/files/9239168/logs.txt)

**Comment by hollodotme** [comment]
Hey @johnadvincula-ada,

### TL;DR 

1. The stream your application tries to read information from broke due to a segmentation fault in the PHP-FPM process that handles your request.

2. You can catch this exception and try to issue the request again. (A "retry" if you will)

3. If 2. does not work, you should try to reproduce the segmentation fault with your specific request data and maybe fix your code or file a bug report at AWS or [the PHP project](https://github.com/php/php-src/issues) - depending on your findings.

### Long version

The underlying cause of this exception is described in your logfile from line 29 to 32:

```log
[01-Aug-2022 09:48:57] WARNING: [pool default] child 892 exited on signal 11 (SIGSEGV - core dumped) after 24.827087 seconds from start
[01-Aug-2022 09:48:57] NOTICE: [pool default] child 900 started
[01-Aug-2022 09:48:57] NOTICE: Terminating ...
[01-Aug-2022 09:48:57] NOTICE: exiting, bye-bye!
```

Those messages originate in the PHP-FPM instance your app is talking to.

The following message indicates a segmentation fault in an active child-process of PHP-FPM causing it to terminate the stream that is used by your application to communicate with that process:

```log
[pool default] child 892 exited on signal 11 (SIGSEGV - core dumped) after 24.827087 seconds from start
```

I can't tell you what the root cause for the this segmentation fault is. This heavily depends on your PHP version, the infrastructure and you application code.

When the checking or reading of a stream fails because it is blocked or terminated (= not existing anymore) from the other end, the fast-cgi-client library is unable to fulfill a request properly. As this behaviour is of course not expected, the library throws the mentioned exception to let you know that something went wrong or is off.

To mitigate this problem you could add a retry-mechanism to your app that catches the exception and tries to re-issue the same request to a new PHP-FPM process. That of course only works if the stream termination / segmentation fault was a temporary issue.

If the issue persists, you should start debugging your app with this specific request to find out what causes the segmentation fault. Depending on your findings, fix your code or file a bug report at AWS or [the PHP project](https://github.com/php/php-src/issues).

Hope this helps.

Cheers, Holger
